execute/plugin_functions.go

```diff
@@ -208,6 +202,117 @@ func filterOutExecutedMessages(
 	return filtered, nil
 }
 
+// truncateObservation truncates the observation to fit within the given maxSize after encoding.
+// It removes data from the observation in the following order:
```

Comment:
I don't think this logic is correct. You should remove messages instead of entire commits, right? Offramp allows partial execution in which only some messages from the CommitRoot are executed, right?

Let's imagine the case in which you have a single, extremely large CommitReport (256 messages) that hits the limit. If you cut off a single CommitReport you will end up producing empty Observation forever, right?

Reply:
https://github.com/smartcontractkit/ccip/blob/80dd5d2b4867ee67c348fc1b05669c015bafa14a/core/services/ocr2/plugins/ccip/ccipexec/ocr2.go#L425

Reply:
The PR description addresses your main comment. For this specific PR we're dealing with large observation with multiple commits. This is a good first line of defense. The task I'm working on currently is if one commit is larger than the size limit. We can't just remove messages easily because of how merkel roots are working so it's a bit tricky and I'm still figuring out how to use hashes instead of messages and pseudo delete a message for example.

simply removing single messages will fail the rest of the flow for now.